src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx

```diff
@@ -0,0 +1,178 @@
+  })
+
+  const expiresIn =
+    Number(hoursTillEnd) < 1 ? `${minutes}m` : `${Math.round(hoursTillEnd)}hr`
+  const offerType = (offers?.edges?.length || []) > 1 ? "Counteroffer" : "Offer"
+
+  let ctaAttributes: {
+    backgroundColor: Color
+    message: string
+    subMessage: string
+    Icon: React.FC<IconProps>
+    url: string
+    modalTitle: string
+  }
+
+  switch (kind) {
+    case "PAYMENT_FAILED": {
+      ctaAttributes = {
+        backgroundColor: "red100",
+        message: "Payment Failed",
+        subMessage:
+          "Unable to process payment for accepted offer. Update payment method.",
+        Icon: AlertCircleFillIcon,
+        url: `/orders/${orderID}/payment/new`,
+        modalTitle: "Update Payment Details",
+      }
+      break
+    }
+    case "OFFER_RECEIVED": {
+      ctaAttributes = {
+        backgroundColor: "copper100",
+        message: `${offerType} Received`,
+        subMessage: `The offer expires in ${expiresIn}`,
+        Icon: AlertCircleFillIcon,
+        url: `/orders/${orderID}`,
+        modalTitle: "Review Offer",
+      }
+      break
+    }
+    case "OFFER_ACCEPTED": {
+      ctaAttributes = {
+        backgroundColor: "green100",
+        message: `Congratulations! ${offerType} Accepted`,
+        subMessage: "Tap to view",
+        Icon: MoneyFillIcon,
+        url: `/orders/${orderID}`,
+        modalTitle: "Offer Accepted",
+      }
+      break
+    }
+    case "OFFER_ACCEPTED_CONFIRM_NEEDED": {
+      ctaAttributes = {
+        backgroundColor: "copper100",
+        message: `Offer Accepted - Confirm total`,
+        subMessage: `The offer expires in ${expiresIn}`,
+        Icon: AlertCircleFillIcon,
+        url: `/orders/${orderID}`,
+        modalTitle: "Review Offer",
+      }
+      break
+    }
+    case "OFFER_RECEIVED_CONFIRM_NEEDED": {
+      ctaAttributes = {
+        backgroundColor: "copper100",
+        message: `Counteroffer Received - Confirm Total`,
+        subMessage: `The offer expires in ${expiresIn}`,
+        Icon: AlertCircleFillIcon,
+        url: `/orders/${orderID}`,
+        modalTitle: "Review Offer",
+      }
+      break
+    }
+    case "PROVISIONAL_OFFER_ACCEPTED": {
+      ctaAttributes = {
+        backgroundColor: "green100",
+        message: `Offer Accepted`,
+        subMessage: "Tap to view",
+        Icon: MoneyFillIcon,
+        url: `/orders/${orderID}`,
+        modalTitle: "Offer Accepted",
+      }
+      break
+    }
+    default: {
+      // this should never happen
+      return null
+    }
+  }
+
+  const {
+    message,
+    subMessage,
+    backgroundColor,
+    Icon,
+    url,
+    modalTitle,
+  } = ctaAttributes
+
+  const navigateToConversation = (path: string, title: string) => {
+    trackEvent(
+      tappedViewOffer({
+        impulse_conversation_id: conversationID,
+        cta: message,
+      })
+    )
+    window.location.assign(path)
+    // figure out how to pass props { orderID, title }
```

Comment:
You should be able to use our router to transition to the conversation more smoothly (in a single page app kind of way, without the hard jump between pages); and to pass `orderID` and `title`, can use queryParams on the url, which you can then pluck off later:  

```tsx
const { router } = useRouter()
... 
router.push(`${path}?orderID=foo&title=bar`)
```

Reply:
Ahhh right, I forgot what the right way to do this was so I went googling